src/main/java/org/jitsi/jicofo/JitsiMeetConferenceImpl.java

```diff
@@ -1543,6 +1545,72 @@ public XMPPError onSessionInfo(JingleSession session, JingleIQ iq)
         return null;
     }
 
+    /**
+     * Handles 'session-terminate' received from the client.
+     *
+     * {@inheritDoc}
+     */
+    @Override
+    public XMPPError onSessionTerminate(JingleSession session, JingleIQ iq)
+    {
+        Participant participant = findParticipantForJingleSession(session);
+
+        // FIXME: (duplicate) there's very similar logic in onSessionAccept/onSessionInfo
+        if (participant == null)
+        {
+            String errorMsg = "No session for " + session.getAddress();
+
+            logger.warn("onSessionInfo: " + errorMsg);
+
+            return XMPPError.from(
+                    XMPPError.Condition.item_not_found, errorMsg).build();
+        }
+
+        BridgeSessionPacketExtension bsPE
+            = iq.getExtension(
+                    BridgeSessionPacketExtension.ELEMENT_NAME,
+                    BridgeSessionPacketExtension.NAMESPACE);
+        String bridgeSessionId = bsPE != null ? bsPE.getId() : null;
+        BridgeSession bridgeSession = findBridgeSession(participant);
+        boolean restartRequested = bsPE != null ? bsPE.isRestart() : false;
+
+        if (bridgeSession == null || !bridgeSession.id.equals(bridgeSessionId))
+        {
+            logger.info(String.format(
+                    "Ignored session-terminate for invalid session: %s, bridge session ID: %s restart: %s",
+                    participant,
+                    bridgeSessionId,
+                    restartRequested));
+
+            return XMPPError.from(XMPPError.Condition.item_not_found, "invalid bridge session ID").build();
+        }
+
+        logger.info(String.format(
+                "Received session-terminate from %s, session: %s, restart: %s",
+                participant,
+                bridgeSession,
+                restartRequested));
+
+        terminateParticipant(participant, null, null);
```

Comment:
Would setting the reason and message to something like "restarting" make sense? It might make debugging easier.

Reply:
Or rather "requested from the client, restartRequested=X"

Reply:
Oh, I see `reason == null` is used as a condition to prevent sending a session-terminate of our own. How about adding a separate parameter?

Reply:
yes, I'll add a separate parameter